src/core/components/hv-route/index.tsx

```diff
@@ -362,8 +362,17 @@ export default function HvRoute(props: Types.Props) {
   const id: string | undefined =
     props.route?.params?.id || navigatorContext.initialRouteName || undefined;
 
+  const state = props.navigation?.getState();
+  const index = state?.index;
+  const type = state?.type;
+  // The nested element is only used when there is an id
+  //    and the navigator is not a stack or is the first screen in a stack
+  //    other stack screens will use a url
+  const includeElement =
+    id && (type !== NavigatorService.NAVIGATOR_TYPE.STACK || index === 0);
```

Comment:
nit: it look like includeElement is used as a boolean, so we might want to explicitly type it, which in turns should require the expression to be reworked as such:
```suggestion
  const includeElement: boolean =
    !!id && (type !== NavigatorService.NAVIGATOR_TYPE.STACK || index === 0);
```

Reply:
I've reworked this bit. I had to move the id out to allow the .get(id) to evaluate properly